src/test/shell/bazel/bazel_symlink_test.sh

```diff
@@ -282,8 +265,7 @@ genrule(
     name="bg",
     srcs=[],
     outs=["bgo"],
-    exec_tools = [":MakeSymlink"],
-    cmd = "$(execpath :MakeSymlink) $@ bad/symlink",
+    cmd = "python3 -c 'import os, sys; os.symlink(*sys.argv[1:])' bad/symlink $@",
```

Comment:
Looking good, thanks! If you can, please add a comment why we are using Python here instead of `ln` so nobody tries to "fix it" in the future.

Reply:
Added the two comments.

Reply:
That unfortunately doesn't work since python3 is not on path for windows machines and we likely don't want to rely on that. That leaves a `py_binary` tool or something like I did here:

```
function test_symlink_created_from_spawn() {
  mkdir a
  cat > a/a.bzl <<'EOF'
def _a_impl(ctx):
    symlink = ctx.actions.declare_symlink(ctx.label.name + ".link")
    output = ctx.actions.declare_file(ctx.label.name + ".file")
    ctx.actions.run_shell(
        outputs = [symlink],
        inputs = depset([]),
        # Fall back to mklink on Windows since Cygwin cannot create danging
        # symlinks.
        # TODO(#15194): Use `command_bat` instead.
        command = r'ln -s "$1" "$2" || "${SYSTEMROOT}"/system32/cmd //C mklink "${2//\\//\\\\}" "${1//\\//\\\\}"',
        arguments = [ctx.attr.link_target, symlink.path],
    )
    ctx.actions.run_shell(
        outputs = [output],
        inputs = depset([symlink]),
        command = "echo input link is $(readlink " + symlink.path + ") > " + output.path,
    )
    return DefaultInfo(files = depset([output]))

a = rule(
    implementation = _a_impl,
    attrs = {
        "link_target": attr.string(),
    }
)
EOF

  cat > a/BUILD <<'EOF'
load(":a.bzl", "a")

a(name="a", link_target="somewhere/over/the/rainbow")
EOF

  bazel build --experimental_allow_unresolved_symlinks //a:a || fail "build failed"
  assert_contains "input link is somewhere/over/the/rainbow" bazel-bin/a/a.file
}
```

We would either need the same for the genrule or we are back to adding a tool of sorts (`py_binary`). Unfortunately `mklink` seems to want paths with `\` not `/`, therefore we cannot easily leverage that there (please see the crazy slash substitution going on there). If you like that, by all means, please copy-paste that and proceed this way. Otherwise, looks like a tool will be necessary (you can just create one out of that python script).

So sorry about that, I approved this change without seeing that the tests failed and changes needed here exceed what I feel comfortable doing with you in the author tag.